prov/verbs/src/verbs_info.c

```diff
@@ -791,21 +794,51 @@ static int vrb_alloc_info(struct ibv_context *ctx, struct fi_info **info,
 
 	switch (ctx->device->transport_type) {
 	case IBV_TRANSPORT_IB:
-		if (ibv_query_gid(ctx, 1, 0, &gid)) {
+		if (ibv_query_gid(ctx, 1, vrb_gl_data.gid_idx, &gid)) {
 			VERBS_INFO_ERRNO(FI_LOG_FABRIC,
 					 "ibv_query_gid", errno);
 			ret = -errno;
 			goto err;
 		}
 
-		name_len = strlen(VERBS_IB_PREFIX) + INET6_ADDRSTRLEN;
-		if (!(fi->fabric_attr->name = calloc(1, name_len + 1))) {
-			ret = -FI_ENOMEM;
+		if (ibv_query_gid_type(ctx, 1, vrb_gl_data.gid_idx, &gid_type)) {
+			VERBS_INFO_ERRNO(FI_LOG_FABRIC,
+					 "ibv_query_gid_type", errno);
+			ret = -errno;
```

Comment:
Rather than failing, we can check the link type and assume either IB or RoCE.  Worst case, we shouldn't fail, but keep the existing behavior and just report as IB.